pkg/sql/executor_ir.go

```diff
@@ -174,6 +176,43 @@ func runSingleSQLFlowStatement(wr *pipe.Writer, sql *parser.SQLFlowStmt, db *dat
 	return r.Execute(submitter)
 }
 
+// RewriteStatementsWithHints combines the hints into the standard SQL(s)
+func RewriteStatementsWithHints(stmts []*parser.SQLFlowStmt, dialect string) []*parser.SQLFlowStmt {
+	hints, sqls := splitHints(stmts, dialect)
+	if len(hints) > 0 {
+		for _, sql := range sqls {
+			if !sql.IsExtendedSyntax() {
+				sql.Original = hints + sql.Original
+			}
+		}
+	}
+	return sqls
+}
+
+func splitHints(stmts []*parser.SQLFlowStmt, dialect string) (string, []*parser.SQLFlowStmt) {
+	hints, sqls := "", []*parser.SQLFlowStmt{}
+	for _, stmt := range stmts {
+		if isHint(stmt, dialect) {
+			hints += stmt.Original
+		} else {
+			sqls = append(sqls, stmt)
+		}
+	}
+	return hints, sqls
+}
+
+func isHint(stmt *parser.SQLFlowStmt, dialect string) bool {
+	if !stmt.IsExtendedSyntax() {
+		if dialect == "alisa" {
```

Comment:
It might be better to use `switch` here and log a warning if `dialect` is not `"alias"`.